packages/bulk-edit/src/browser/bulk-edit-tree/bulk-edit-tree.ts

```diff
@@ -0,0 +1,105 @@
+/********************************************************************************
+ * Copyright (c) 2020 SAP SE or an SAP affiliate company and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable } from 'inversify';
+import { TreeNode, CompositeTreeNode, SelectableTreeNode, ExpandableTreeNode, TreeImpl } from '@theia/core/lib/browser';
+import { UriSelection } from '@theia/core/lib/common/selection';
+import { BulkEditNodeSelection } from './bulk-edit-node-selection';
+import URI from '@theia/core/lib/common/uri';
+
+@injectable()
+export class BulkEditTree extends TreeImpl {
+    public async initTree(workspaceEdit: monaco.languages.WorkspaceEdit, fileContents: Map<string, string>): Promise<void> {
+        this.root = <CompositeTreeNode>{
+            visible: false,
+            id: 'theia-bulk-edit-tree-widget',
+            name: 'BulkEditTree',
+            children: this.getChildren(workspaceEdit, fileContents),
+            parent: undefined
+        };
+    }
+
+    private getChildren(workspaceEdit: monaco.languages.WorkspaceEdit, fileContentsMap: Map<string, string>): BulkEditInfoNode[] {
+        let bulkEditInfos: BulkEditInfoNode[] = [];
+        if (workspaceEdit.edits) {
+            bulkEditInfos = workspaceEdit.edits
+                .map(edit =>
+                    (edit && ('resource' in edit) && edit.resource && edit.resource.path) ? edit.resource.path :
+                        ((edit && ('newUri' in edit) && edit.newUri && edit.newUri.path) ? edit.newUri.path : undefined))
+                .filter((path, index, arr) => path && arr.indexOf(path) === index)
```

Comment:
Won't this always be `true`?
```ts
arr.indexOf(path) === index)
```

Reply:
this line creates distinct paths:
![image](https://user-images.githubusercontent.com/71069170/99272393-6c924a80-2830-11eb-8047-d1661e19190a.png)

Reply:
@vince-fugnitto Updated to:
private getResourcePath(edit: monaco.languages.WorkspaceTextEdit | monaco.languages.WorkspaceFileEdit): string | undefined {
        return WorkspaceTextEdit.is(edit) ? edit.resource.path : WorkspaceFileEdit.is(edit) && edit.newUri ? edit.newUri.path : undefined;
    }